client/src/main/java/io/pravega/client/segment/impl/SegmentOutputStreamImpl.java

```diff
@@ -336,14 +355,14 @@ public void processingFailure(Exception error) {
             state.failConnection(error);
         }
     }
-    
+
     /**
-     * @see SegmentOutputStream#write(java.nio.ByteBuffer,
-     *      java.util.concurrent.CompletableFuture)
+     * @see SegmentOutputStream#write(PendingEvent)
      */
     @Override
     @Synchronized
     public void write(PendingEvent event) throws SegmentSealedException {
+        checkState(!state.isAlreadySealed(), "Segment: {} is already sealed", segmentName);
```

Comment:
I think we might want to take a "belt and suspenders approach" and change the callback to be more general and pass to it a list of unsent events. Then leave open the possibility that the callback could be invoked more than once. This would let us avoid any race conditions and more easily reason about events not being lost and order being maintained.